pkg/tsdb/cloudwatch/time_series_query_test.go

```diff
@@ -136,3 +136,224 @@ func TestTimeSeriesQuery(t *testing.T) {
+		return datasourceInfo{}, nil
+	})
+	executor := newExecutor(im, newTestConfig(), &fakeSessionCache{})
+
+	t.Run("where user defines search expression and alias is defined, then frame name prioritizes period and stat from expression over input", func(t *testing.T) {
+		query := newTestQuery(t, queryParameters{
+			MetricEditorMode: MetricEditorModeRaw, // contributes to isUserDefinedSearchExpression = true
+			Alias:            "{{period}} {{stat}}",
+			Expression:       `SEARCH('{AWS/EC2,InstanceId} MetricName="CPUUtilization"', 'Average', 300)`, // period 300 and stat 'Average' parsed from this expression
+			Statistic:        "Maximum",                                                                    // stat parsed from expression takes precedence over 'Maximum'
+			Period:           "1200",                                                                       // period parsed from expression takes precedence over 1200
+		})
+
+		resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+			PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+			Queries: []backend.DataQuery{
+				{
+					RefID:     "A",
+					TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+					JSON:      query,
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "300 Average", resp.Responses["A"].Frames[0].Name)
+	})
+
+	t.Run("where no alias is provided and query is math expression, then frame name is queryId", func(t *testing.T) {
+		query := newTestQuery(t, queryParameters{
+			MetricEditorMode: MetricEditorModeRaw,
+		})
+
+		resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+			PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+			Queries: []backend.DataQuery{
+				{
+					RefID:     "A",
+					TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+					JSON:      query,
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "query id", resp.Responses["A"].Frames[0].Name)
+	})
+
+	t.Run("where no alias provided and query type is MetricQueryTypeQuery, then frame name is label", func(t *testing.T) {
+		query := newTestQuery(t, queryParameters{
+			MetricQueryType: MetricQueryTypeQuery,
+		})
+
+		resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+			PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+			Queries: []backend.DataQuery{
+				{
+					RefID:     "A",
+					TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+					JSON:      query,
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "response label", resp.Responses["A"].Frames[0].Name)
```

Comment:
for this and the previous case, maybe it would be more readable if the expected value - `"response label"` - was put in a variable that was also passed as param to newTestQuery? So reader doesn't have to scroll up to that func. Up to you!

Reply:
yup it's done now